searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h

```diff
@@ -82,6 +83,8 @@ struct IDocumentMetaStore : public search::IDocumentMetaStore,
      */
     virtual void compactLidSpace(DocId wantedLidLimit) = 0;
 
+    virtual void set_operation_listener(documentmetastore::OperationListener::SP op_listener) = 0;
```

Comment:
Consider forward declaring documentmetastore::OperationListener and using std::shared_ptr here.

Reply:
Fixed